### PR TITLE
collectd: clean up building with custom plugins

### PR DESCRIPTION
**Pull request by astro**
###### Motivation for this change

Just refreshing and cleaning up the customizability of collectd that I introduced in #71406.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/system/collectd/plugins.nix [hunk 1]
```diff
@@ -46,299 +49,90 @@ let
...
-    zfs_arc = {};
-    zone = {};
-    zookeeper = {};
+    amqp.buildInputs = [
+      yajl
+    ] ++ lib.optionals stdenv.isLinux [ rabbitmq-c ];
+    apache.buildInputs = [ curl ];
+    ascent.buildInputs = [ curl libxml2 ];
+    battery.buildInputs = lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.IOKit
```
Comment (SuperSandro2000) [inline pkgs/tools/system/collectd/plugins.nix:58]:
```suggestion
      IOKit
```
Please do ``imherit (darwin.apple_sdk.frameworks) IOKit`` in the top level call.

File: pkgs/tools/system/collectd/plugins.nix [hunk 2]
```diff
@@ -46,299 +49,90 @@ let
...
+    bind.buildInputs = [ curl libxml2 ];
+    ceph.buildInputs = [ yajl ];
+    curl.buildInputs = [ curl ];
+    curl_json.buildInputs = [ curl yajl ];
+    curl_xml.buildInputs = [ curl libxml2 ];
+    dbi.buildInputs = [ libdbi ];
+    disk.buildInputs = lib.optionals stdenv.isLinux [
+      udev
+    ] ++ lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.IOKit
```
Comment (SuperSandro2000) [inline pkgs/tools/system/collectd/plugins.nix:69]:
```suggestion
      IOKit
```

**Comment by astro** [comment]
@SuperSandro2000 Thanks for the review.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 134352` at 3f722742 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>collectd</li>
    <li>collectd-data</li>
  </ul>
</details>

**Comment by astro** [comment]
I've just added the new OpenIPMI input, then rebased the PR.

**Comment by SuperSandro2000** [comment]
@astro maybe you want to add yourself as a maintainer? Then it would be easier to quickly merge further changes in the future.

**Pull request merged**